pkg/queue/rethinkdb/worker.go

```diff
@@ -0,0 +1,114 @@
+package rethinkdb
+
+import (
+	"time"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/aquasecurity/harbor-scanner-trivy/pkg/etc"
+	"github.com/aquasecurity/harbor-scanner-trivy/pkg/queue"
+	"github.com/aquasecurity/harbor-scanner-trivy/pkg/scan"
+
+	r "gopkg.in/rethinkdb/rethinkdb-go.v6"
+)
+
+type workerPool struct {
+	conn           r.QueryExecutor
+	started        bool
+	done           chan bool
+	scanController scan.Controller
+	cfg            etc.Rethink
+}
+
+func NewWorkerPool(conn r.QueryExecutor, scanController scan.Controller, rethinkCfg etc.Rethink) queue.Worker {
```

Comment:
Any reason why `scanController` passed in here as opposed to generating it in the struct like redis?

Reply:
Well, simply because it is a dependency of the worker. I don't see a reason why the Redis implementation doesn't do it. It has a number of drawbacks:
1. Every time a job is started a new instance of `scanController` is generated when, in practice, a single instance can be reused (Singleton).
2. The implementation is coupled to the concrete `scanController` instead of depending on an abstraction (the interface).
3. The method cannot be tested.